audio-output/setSinkId-manual.https.html

```diff
@@ -0,0 +1,132 @@
+<p>Available but unauthorized devices (only those for which we can gain permission can be selected):</p>
+<ul id="available"></ul>
+<p>Authorized  devices:</p>
+<ul id="authorized"></ul>
+<audio controls id="beat" src="/media/sound_5.mp3" loop></audio>
+
+<div id='log'></div>
+<script src=/resources/testharness.js></script>
+<script src=/resources/testharnessreport.js></script>
+<script>
+"use strict";
+
+const is_output = d => d.kind === "audiooutput";
+const by_id = (id) => d => d.groupId === id;
+const is_input = d => d.kind === "audioinput";
+const audio = document.getElementById("beat");
+const available = document.getElementById("available");
+const authorized = document.getElementById("authorized");
+
+let outputList;
+
+const selectDeviceTester = (t) => (e) => {
+    const groupId = e.target.dataset["groupid"];
+    const device = outputList.find(by_id(groupId));
+    if (audio.paused) audio.play();
+    promise_test(pt => audio.setSinkId(device.deviceId).then(() => {
+        assert_equals(device.deviceId, audio.sinkId);
+
+        const pass = document.createElement("button");
+        const fail = document.createElement("button");
+
+        const result = (bool) => () => {
+            assert_true(bool, "Sound rendered on right device");
+            fail.remove();
+            pass.remove();
+            audio.pause();
+            e.target.checked = false;
+            e.target.disabled = true;
+            t.done();
+        };
+
+        pass.style.backgroundColor = "#0f0";
+        pass.textContent = "\u2713 Sound plays on " + device.label;
+        pass.addEventListener("click", result(true));
+
+        fail.style.backgroundColor = "#f00";
+        fail.textContent = "\u274C Sound doesn't play on " + device.label;
+        fail.addEventListener("click", result(true));
+
+        const container = e.target.parentNode.parentNode;
+        container.appendChild(pass);
+        container.appendChild(fail);
+    }), "setSinkId for " + device.label + " resolves");
+};
+
+const addAuthorizedDevice = (groupId) => {
+    const device = outputList.find(by_id(groupId));
+    const async_t = async_test("Selecting output device " + device.label + " makes the audio rendered on the proper device");
+    const li = document.createElement("li");
+    const label = document.createElement("label");
+    const input = document.createElement("input");
+    input.type = "radio";
+    input.name = "device";
+    input.dataset["groupid"] = device.groupId;
+    input.addEventListener("change", selectDeviceTester(async_t));
+    const span = document.createElement("span");
+    span.textContent = device.label;
+    label.appendChild(input);
+    label.appendChild(span);
+    li.appendChild(label);
+    authorized.appendChild(li);
+};
+
+const authorizeDeviceTester = (t) => (e) => {
+    const groupId = e.target.dataset["groupid"];
+    navigator.mediaDevices.getUserMedia({audio: {groupId}})
```

Comment:
I'm not sure that getting permission with getUserMedia should be part of the WPT test for setSinkId. 
In Chrome, gUM authorizes output devices although I see that as a Chrome-specific implementation detail. It's certainly not something in any spec. 
The spec-compliant way to get permission to use the device would be to use the permissions API, but that does not work in Chrome as of now.

Reply:
The spec says
>  implementations MUST support implicit consent via the getUserMedia() permission prompt; when an audio input device is authorized and opened via getUserMedia(), this also authorizes access to any associated audio output devices (i.e., those with the same groupId)
so testing it seems reasonable to me

Reply:
You are correct. 
My concern was about output devices that have no associated microphone and are thus not required to be authorized by getUserMedia, but there is currently no standard programmatic way to authorize those anyway, although Chrome authorizes them as well with gUM. I thought the permissions API would allow a standard way for those, but I just found out that's not the case.
So, this test LGTM.